samples/client/petstore/r_test/R/PetApi.r

```diff
@@ -0,0 +1,161 @@
+# Swagger Petstore
+# 
+# This is a sample server Petstore server.  You can find out more about Swagger at [http://swagger.io](http://swagger.io) or on [irc.freenode.net, #swagger](http://swagger.io/irc/).  For this sample, you can use the api key `special-key` to test the authorization filters.
+# 
+# OpenAPI spec version: 1.0.0
+# Contact: apiteam@swagger.io
+# Generated by: https://github.com/swagger-api/swagger-codegen.git
+
+PetApi <- R6::R6Class(
+  'PetApi',
+  public = list(
+    userAgent = "Swagger-Codegen/1.0.0/r",
+    host = NULL,
+    basePath = NULL,
+    scheme = NULL,
+    url = NULL,
+    initialize = function(host, basePath, scheme){
+      self$host <- host
+      self$basePath <- basePath
+      self$scheme <- scheme
+      self$url <- sprintf("%s://%s/%s/pet/", scheme, host, basePath)
+    },
+
+    add_pet = function(body){
+      resp <- httr::POST(paste0(self$url),
+          httr::add_headers("User-Agent" = self$userAgent, "accept" = "application/json", "content-type" = "application/xml")
+          ,body = body$toJSON()
+          )
+
+      if (httr::status_code(resp) >= 200 && httr::status_code(resp) <= 299) {
+        # void response, no need to return anything
+      } else if (httr::status_code(resp) >= 400 && httr::status_code(resp) <= 499){
+        Response$new("API client error", resp)
+      } else if (httr::status_code(resp) >= 500 && httr::status_code(resp) <= 599){
+        Response$new("API server error", resp)
+      }
+
+    },
+    delete_pet = function(pet_id, api_key){
+      resp <- httr::DELETE(paste0(self$url, pet_id),
+          httr::add_headers("User-Agent" = self$userAgent, "content-type" = "application/xml", "api_key" = api_key)
+          )
+
+      if (httr::status_code(resp) >= 200 && httr::status_code(resp) <= 299) {
+        # void response, no need to return anything
+      } else if (httr::status_code(resp) >= 400 && httr::status_code(resp) <= 499){
+        Response$new("API client error", resp)
+      } else if (httr::status_code(resp) >= 500 && httr::status_code(resp) <= 599){
+        Response$new("API server error", resp)
+      }
+
+    },
+    find_pets_by_status = function(status){
+      resp <- httr::GET(paste0(self$url),
+          httr::add_headers("User-Agent" = self$userAgent, "content-type" = "application/xml")
+          )
+      # TODO add support for query parameters "status"
+
+      if (httr::status_code(resp) >= 200 && httr::status_code(resp) <= 299) {
+        parsed <- jsonlite::fromJSON(httr::content(resp, "text", encoding = "UTF-8"),
+                                 simplifyVector = FALSE)
+        result <- Pet$fromJSON(parsed)
```

Comment:
@ramnov is there any way we can provide a `fromJSON` method to construct the object based on a JSON string?

For other languages, there are libraries for (de)serialization between JSON and object. I wonder if R has similar package.

Reply:
There is no package in R to deserialize/serialize between R6 object and JSON. fromJSON() and toJSON() methods have to be implemented by ourselves in R6 Classes. I believe Pet.R already has a toJSON() method. Can you add this fromJSON() method as well to Pet.R in Pet Class : 

fromJSON = function(petJson) {
      petObject <- jsonlite::fromJSON(petJson)
      self$id <- petObject$id
      self$category <- Element$new(petObject$category$id, petObject$category$name)
      self$name <- petObject$name
      self$photoUrls <- petObject$photoUrls
      self$tags <- lapply(petObject$tags, function(x) Element$new(x$id, x$name))
      self$status <- petObject$status
    } 

And then this function can be used in PetApi.R like this : 

result <- Pet$new()$fromJSON(httr::content(resp, "text", encoding = "UTF-8"),
                                 simplifyVector = FALSE)